app/src/NativeBase/Components/Forms/TextInputControl.tsx

```diff
@@ -0,0 +1,144 @@
+/**
+ * @file Form input.
+ */
+
+import React, { FC } from 'react'
+import { Input, Box, Text } from 'native-base'
+import ErrorMessage from './ErrorMessage'
+
+interface TextInputControlProps {
+  error: boolean
+  required: boolean
+  errorType?: 'invalid' | 'missing'
+  label: string
+  onBlur?: () => void
+  onChange: (value: string) => void
+  type: 'email' | 'firstName' | 'lastName'
+  value: string
+}
+
+const TextInputControl: FC<TextInputControlProps> = ({
+  error,
+  errorType,
+  label,
+  onBlur,
+  onChange,
+  required,
+  type,
+  value,
+}) => {
+  let autoCapitalize = 'none' as
+    | 'none'
+    | 'sentences'
+    | 'words'
+    | 'characters'
+    | undefined
+  let autoComplete = 'off' as
+    | 'birthdate-day'
+    | 'birthdate-full'
+    | 'birthdate-month'
+    | 'birthdate-year'
+    | 'cc-csc'
+    | 'cc-exp'
+    | 'cc-exp-day'
+    | 'cc-exp-month'
+    | 'cc-exp-year'
+    | 'cc-number'
+    | 'email'
+    | 'gender'
+    | 'name'
+    | 'name-family'
+    | 'name-given'
+    | 'name-middle'
+    | 'name-middle-initial'
+    | 'name-prefix'
+    | 'name-suffix'
+    | 'password'
+    | 'password-new'
+    | 'postal-address'
+    | 'postal-address-country'
+    | 'postal-address-extended'
+    | 'postal-address-extended-postal-code'
+    | 'postal-address-locality'
+    | 'postal-address-region'
+    | 'postal-code'
+    | 'street-address'
+    | 'sms-otp'
+    | 'tel'
+    | 'tel-country-code'
+    | 'tel-national'
+    | 'tel-device'
+    | 'username'
+    | 'username-new'
+    | 'off'
+    | undefined
+  let textContentType = 'none' as
+    | 'none'
+    | 'URL'
+    | 'addressCity'
+    | 'addressCityAndState'
+    | 'addressState'
+    | 'countryName'
+    | 'creditCardNumber'
+    | 'emailAddress'
+    | 'familyName'
+    | 'fullStreetAddress'
+    | 'givenName'
+    | 'jobTitle'
+    | 'location'
+    | 'middleName'
+    | 'name'
+    | 'namePrefix'
+    | 'nameSuffix'
+    | 'nickname'
+    | 'organizationName'
+    | 'postalCode'
+    | 'streetAddressLine1'
+    | 'streetAddressLine2'
+    | 'sublocality'
+    | 'telephoneNumber'
+    | 'username'
+    | 'password'
+    | 'newPassword'
+    | 'oneTimeCode'
+    | undefined
+  const textInputStyle = { fontSize: 16 }
+
+  switch (type) {
+    case 'email':
+      autoComplete = 'email'
+      textContentType = 'emailAddress'
+      break
+    case 'firstName':
+      autoCapitalize = 'words'
+      autoComplete = 'name-given'
+      textContentType = 'givenName'
+      break
+    case 'lastName':
+      autoCapitalize = 'words'
+      autoComplete = 'name-family'
+      textContentType = 'familyName'
+      break
+  }
+
+  return (
+    <Box>
+      <Text fontWeight="600" fontSize="lg">
+        {label}
+        {required ? <Text color="error.100">*</Text> : ''}
+      </Text>
+      <Input
```

Comment:
Something I hadn't thought about is accessibility.  From looking at the NativeBase docs, I think we should be using [FormControl](https://docs.nativebase.io/form-control) here (and that allows you to set an `isRequired` property too), and `<FormControl.Label>` above instead of `<Text>`, and use `<FormControl.ErrorMessage>` (see examples in the docs).

I need to do some of this myself on the PR I've just submitted! 🙂